### PR TITLE
Fix plugin autoload, use pip to install package during build, add Windows build

### PR DESCRIPTION
**Pull request by tongyuantongyu**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

This PR contains 3 fixes & updates. If you prefer I can break them into separate PRs.

- Fix plugin autoload:
`conda` patches executable binary to adapt to installation prefix path, but compiler optimization will inline the length of the system plugin path, causing errors. This PR fixes it by prevent such optimization.

- Use pip to install package during build:
Use pip to install the package creates `dist-info`, so that `pip` can recognize the installed `VapourSynth` package, and not trying to install the pypi version in case user installs dependent packages using pip.

- Add Windows build
Add Windows build script. This requires fixed `zimg` package from https://github.com/conda-forge/zimg-feedstock/pull/7 .

### CODE REVIEW

**Comment by tongyuantongyu** [comment]
@conda-forge-admin please rerender

**Comment by tongyuantongyu** [comment]
@conda-forge-admin please rerender

**Comment by tongyuantongyu** [comment]
The failure in Windows build is due to wrong link name of zimg.

File: recipe/meta.yaml
Comment (sshockwave) [inline recipe/meta.yaml:51]:
```suggestion
```

`compiler('c')` has this, as I recall

File: recipe/patches/script-fix-python-path.patch
Comment (sshockwave) [inline recipe/patches/script-fix-python-path.patch:11]:
```suggestion
+    <IncludePath>../../include;$(LIBRARY_INC);$(IncludePath)</IncludePath>
```
I'm not sure if this works. Please review.

File: recipe/meta.yaml
Comment (sshockwave) [inline recipe/meta.yaml:37]: ?

Comment (sshockwave) [reply recipe/meta.yaml:37]: `"${PYTHON}"`?

**Comment by sshockwave** [comment]
Thank you for all the work! I'll have another review after this batch.

Would you like to be a maintainer as well? Simply add your ID in `meta.yaml`.

**Review by sshockwave — CHANGES_REQUESTED** [review]

File: recipe/bld.bat
Comment (tongyuantongyu) [reply recipe/bld.bat:22]: This is intended. This script is doing a "portable" install (in VapourSynth's term) so that VapourSynth won't touch Appdata or Registry, which can break environment isolation.
In a "portable" install, the place of `VapourSynth.dll` decides where to search for plugins, and exist VapourSynth users are used to placing plugins in `%PREFIX%/vapoursynth64/plugins`.
Likely in "portable" install `VSScript.dll` tries to load `python%CONDA_PY%.dll` from the same path, so it is placed in `%PREFIX%` as well.

File: recipe/patches/script-fix-python-path.patch
Comment (tongyuantongyu) [reply recipe/patches/script-fix-python-path.patch:11]: This is to include `Python.h` which is in `%PREFIX%\include` instead of `%LIBRARY_INC%` (%PREFIX%\Library\include).

File: recipe/meta.yaml
Comment (tongyuantongyu) [reply recipe/meta.yaml:46]: Done

File: recipe/bld.bat
Comment (tongyuantongyu) [reply recipe/bld.bat]: Done

File: recipe/meta.yaml
Comment (tongyuantongyu) [reply recipe/meta.yaml]: Done. Without this there are some warnings during DSO check. But installing `python` installs `ucrt` anyway so I believe it will be fine.

File: recipe/meta.yaml
Comment (tongyuantongyu) [reply recipe/meta.yaml:37]: Not quite this, but done.

**Comment by tongyuantongyu** [comment]
The just-released autoconf 2.71 seems broken: https://github.com/conda-forge/autoconf-feedstock/issues/38 .
Maybe force `autoconf != 2.71` in the recipe for now?

**Comment by sshockwave** [comment]
> Maybe force `autoconf != 2.71` in the recipe for now?

:sweat_smile:

Sure, add `autoconf <2.71`

File: recipe/meta.yaml
Comment (sshockwave) [reply recipe/meta.yaml]: I just checked the logs and I couldn't find DSO-related warnings. Could you please take another look?

File: recipe/bld.bat
Comment (sshockwave) [reply recipe/bld.bat:22]: > VapourSynth users are used to placing plugins in %PREFIX%/vapoursynth64/plugins

Sorry for not being familiar with the VapourSynth ecosystem, but what does `%PREFIX%` refer to outside conda? Since plugins are usually a bunch of `.dll`s (or `.py`), can't we put the entire `vapoursynth64` inside `%LIBRARY_BIN%`? The plugins dir solely depends on `<VapourSynth.dll path>\vapoursynth64\plugins` as per the [doc](https://www.vapoursynth.com/doc/installation.html).

Anyway, I totally understand that this is just a matter of preference. I just trying to do things the right way.

**Comment by sshockwave** [comment]
I'm curious, do you happen to know about the course every Friday~Sunday night?

**Comment by tongyuantongyu** [comment]
> Out of curiosity, do you happen to know about the course every Friday~Sunday night?

I don't know. What's that?

**Comment by sshockwave** [comment]
@tongyuantongyu  It's a private interest group and it seems that you are not there. Pardon me for bothering you :stuck_out_tongue_closed_eyes:

File: recipe/meta.yaml
Comment (tongyuantongyu) [reply recipe/meta.yaml]: Maybe some differences between my local environment and the build CI. Since it's fine in the CI it should be OK.

File: recipe/bld.bat
Comment (sshockwave) [reply recipe/bld.bat:22]: I see. Thank you for your detailed explanation. Indeed, I don't see many use cases for linking VapourSynth though it claims to be a C++ library. We'll get into that when that actually happens.

**Pull request merged**
